File: app/src/main/java/zut/edu/cn/notepad/activity/SettingActivity.java
package zut.edu.cn.notepad.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.KeyEvent;
import android.view.Window;

import java.util.Objects;

import zut.edu.cn.notepad.R;
import zut.edu.cn.notepad.cursom.TitleBar;


public class SettingActivity extends AppCompatActivity {

    private TitleBar titleBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        Objects.requireNonNull(getSupportActionBar()).hide();
        setContentView(R.layout.activity_setting);
        init();
        initView();
    }

    private void Intent_Activity() {
        startActivity(new Intent(SettingActivity.this, MainActivity.class));
        finish();
        overridePendingTransition(R.anim.fade_in, R.anim.fade_out);
    }

    private void initView() {
        titleBar.setLeftImageResource(R.drawable.ic_arrow_back_black_24dp);
        titleBar.setLeftClickListener(v -> Intent_Activity());


    }

    private void init() {
        titleBar = findViewById(R.id.title_bar);

    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (keyCode == KeyEvent.KEYCODE_BACK) {
            Intent_Activity();
            return true;
        }
        return super.onKeyDown(keyCode, event);
    }
}

File: app/src/main/java/wooyun/esnb/activity/SettingActivity.kt
package wooyun.esnb.activity

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.view.Window
import wooyun.esnb.R
import wooyun.esnb.fragment.SettingFragment
import wooyun.esnb.view.TitleBar
import java.util.*

@Suppress("NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS")
class SettingActivity : AppCompatActivity() {
    private var titleBar: TitleBar? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        Objects.requireNonNull(supportActionBar)!!.hide()
        setContentView(R.layout.activity_setting)
        init()
        initView()
    }

    private fun Intent_Activity() {
        startActivity(Intent(this@SettingActivity, MainActivity::class.java))
        finish()
        overridePendingTransition(R.anim.fade_in, R.anim.fade_out)
    }

    private fun initView() {
        titleBar!!.setLeftImageResource(R.drawable.ic_arrow_back_black_24dp)
        titleBar!!.setLeftClickListener(View.OnClickListener { v: View? -> Intent_Activity() })
    }

    private fun init() {
        titleBar = findViewById(R.id.title_bar)
        supportFragmentManager
                .beginTransaction()
                .add(R.id.fragment, SettingFragment(), null)
                .commit()
    }


    @SuppressLint("CommitPrefEdits")
    override fun onBackPressed() {
        super.onBackPressed()
//        val sharedPreferences = Tools().spGet(this, "wooyun.notepad_preferences")
//        if (!Str.isUrl(sharedPreferences.getString("edit_pt", ""))) {
//            sharedPreferences.edit().putString("edit_pt", "")
//        }
        Intent_Activity()
    }

    override fun onDestroy() {
        super.onDestroy()
        finish()
    }
}
